x-pack/plugins/canvas/server/routes/workpad.js

```diff
@@ -5,13 +5,20 @@
  */
 
 import boom from 'boom';
-import { CANVAS_TYPE, API_ROUTE_WORKPAD } from '../../common/lib/constants';
+import {
+  CANVAS_TYPE,
+  API_ROUTE_WORKPAD,
+  API_ROUTE_WORKPAD_ASSETS,
+  API_ROUTE_WORKPAD_STRUCTURES,
+} from '../../common/lib/constants';
 import { getId } from '../../public/lib/get_id';
 
 export function workpad(server) {
   //const config = server.config();
   const { errors: esErrors } = server.plugins.elasticsearch.getCluster('data');
   const routePrefix = API_ROUTE_WORKPAD;
+  const routePrefixAssets = API_ROUTE_WORKPAD_ASSETS;
+  const routePrefixStructures = API_ROUTE_WORKPAD_STRUCTURES;
```

Comment:
nit: I don't know if you explored this at all, but doing some kind of detection in a single route would be nice. Having 3 different routes to update a workpad (just the workpad, just the assets, or both) is easy, but seems like overkill. I'm fine leaving it this way though, just wanted to express that opinion.